diff --git a/dist/index.cjs b/dist/index.cjs
index 4a57b130030d097fef8a47041b13ee264c8c2753..d9944fd5ecb4c09049a6f269f1118983a795a15f 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -9190,7 +9190,8 @@ function runHandlers(map, event, view, scope) {
             // Ctrl-Alt may be used for AltGr on Windows
             !(browser.windows && event.ctrlKey && event.altKey) &&
             // Alt-combinations on macOS tend to be typed characters
-            !(browser.mac && event.altKey && !(event.ctrlKey || event.metaKey)) &&
+            // Allow Alt+Shift combinations to be treated as shortcuts rather than special character input
+            !(browser.mac && event.altKey && !event.shiftKey && !(event.ctrlKey || event.metaKey)) &&
             (baseName = w3cKeyname.base[event.keyCode]) && baseName != name) {
             if (runFor(scopeObj[prefix + modifiers(baseName, event, true)])) {
                 handled = true;
diff --git a/dist/index.d.cts b/dist/index.d.cts
index 0f9bebef8172e0661495b80be58120c49d4feb7b..0205eb6b5d6fcd3897bc58f4b2abd2bf88fe973a 100644
--- a/dist/index.d.cts
+++ b/dist/index.d.cts
@@ -520,7 +520,7 @@ declare class ViewPlugin<V extends PluginValue, Arg = undefined> {
         new (view: EditorView, arg: Arg): V;
     }, spec?: PluginSpec<V>): ViewPlugin<V, Arg>;
 }
-interface MeasureRequest<T> {
+export interface MeasureRequest<T> {
     /**
     Called in a DOM read phase to gather information that requires
     DOM layout. Should _not_ mutate the document.
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 0f9bebef8172e0661495b80be58120c49d4feb7b..0205eb6b5d6fcd3897bc58f4b2abd2bf88fe973a 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -520,7 +520,7 @@ declare class ViewPlugin<V extends PluginValue, Arg = undefined> {
         new (view: EditorView, arg: Arg): V;
     }, spec?: PluginSpec<V>): ViewPlugin<V, Arg>;
 }
-interface MeasureRequest<T> {
+export interface MeasureRequest<T> {
     /**
     Called in a DOM read phase to gather information that requires
     DOM layout. Should _not_ mutate the document.
diff --git a/dist/index.js b/dist/index.js
index cea6116972b9276e2859746334769abbcce6cd9c..71962fee3cbe7537e78eec0df19e37d726237f8e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -9185,7 +9185,8 @@ function runHandlers(map, event, view, scope) {
             // Ctrl-Alt may be used for AltGr on Windows
             !(browser.windows && event.ctrlKey && event.altKey) &&
             // Alt-combinations on macOS tend to be typed characters
-            !(browser.mac && event.altKey && !(event.ctrlKey || event.metaKey)) &&
+            // Allow Alt+Shift combinations to be treated as shortcuts rather than special character input
+            !(browser.mac && event.altKey && !event.shiftKey && !(event.ctrlKey || event.metaKey)) &&
             (baseName = base[event.keyCode]) && baseName != name) {
             if (runFor(scopeObj[prefix + modifiers(baseName, event, true)])) {
                 handled = true;
