diff --git a/dist/index.cjs b/dist/index.cjs
index 0fb53bd177b2ba028c445574bc40134f3686e555..7fde56e7ab7c3ff2f1ae783eeb6259c8594a2bff 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -9204,7 +9204,8 @@ function runHandlers(map, event, view, scope) {
             // Ctrl-Alt may be used for AltGr on Windows
             !(browser.windows && event.ctrlKey && event.altKey) &&
             // Alt-combinations on macOS tend to be typed characters
-            !(browser.mac && event.altKey && !(event.ctrlKey || event.metaKey)) &&
+            // Allow Alt+Shift combinations to be treated as shortcuts rather than special character input
+            !(browser.mac && event.altKey && !event.shiftKey && !(event.ctrlKey || event.metaKey)) &&
             (baseName = w3cKeyname.base[event.keyCode]) && baseName != name) {
             if (runFor(scopeObj[prefix + modifiers(baseName, event, true)])) {
                 handled = true;
diff --git a/dist/index.d.cts b/dist/index.d.cts
index 0f9bebef8172e0661495b80be58120c49d4feb7b..0205eb6b5d6fcd3897bc58f4b2abd2bf88fe973a 100644
--- a/dist/index.d.cts
+++ b/dist/index.d.cts
@@ -520,7 +520,7 @@ declare class ViewPlugin<V extends PluginValue, Arg = undefined> {
         new (view: EditorView, arg: Arg): V;
     }, spec?: PluginSpec<V>): ViewPlugin<V, Arg>;
 }
-interface MeasureRequest<T> {
+export interface MeasureRequest<T> {
     /**
     Called in a DOM read phase to gather information that requires
     DOM layout. Should _not_ mutate the document.
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 0f9bebef8172e0661495b80be58120c49d4feb7b..0205eb6b5d6fcd3897bc58f4b2abd2bf88fe973a 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -520,7 +520,7 @@ declare class ViewPlugin<V extends PluginValue, Arg = undefined> {
         new (view: EditorView, arg: Arg): V;
     }, spec?: PluginSpec<V>): ViewPlugin<V, Arg>;
 }
-interface MeasureRequest<T> {
+export interface MeasureRequest<T> {
     /**
     Called in a DOM read phase to gather information that requires
     DOM layout. Should _not_ mutate the document.
diff --git a/dist/index.js b/dist/index.js
index 2507607388dd6f50deabe6b13133a3583103ac11..0ba959432970c3fabad30ddbadf9f47478e21ee8 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -9199,7 +9199,8 @@ function runHandlers(map, event, view, scope) {
             // Ctrl-Alt may be used for AltGr on Windows
             !(browser.windows && event.ctrlKey && event.altKey) &&
             // Alt-combinations on macOS tend to be typed characters
-            !(browser.mac && event.altKey && !(event.ctrlKey || event.metaKey)) &&
+            // Allow Alt+Shift combinations to be treated as shortcuts rather than special character input
+            !(browser.mac && event.altKey && !event.shiftKey && !(event.ctrlKey || event.metaKey)) &&
             (baseName = base[event.keyCode]) && baseName != name) {
             if (runFor(scopeObj[prefix + modifiers(baseName, event, true)])) {
                 handled = true;
