diff --git a/dist/browser.d.mts b/dist/browser.d.mts
new file mode 100644
index 0000000000000000000000000000000000000000..cc05bfacd082570e90b60079c4c986b30346c225
--- /dev/null
+++ b/dist/browser.d.mts
@@ -0,0 +1,2 @@
+export { default } from './index.mjs';
+export * from './index.mjs';
diff --git a/dist/browser.d.ts b/dist/browser.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..15cea0ac44c79d484742fa5d92777185d6ffbfa4
--- /dev/null
+++ b/dist/browser.d.ts
@@ -0,0 +1,2 @@
+export { default } from './index.js';
+export * from './index.js';
