diff --git a/lib/inline.js b/lib/inline.js
index c40534e..8fd7584 100644
--- a/lib/inline.js
+++ b/lib/inline.js
@@ -132,7 +132,10 @@ export default function makeJuiceClient(juiceClient) {
               ? decode(styleAttributeValue)
               : styleAttributeValue;
             const cssText = '* { ' + cssStyleAttributeValue + ' } ';
-            addProps(utils.parseCSS(cssText, { strict: true })[0][1], new utils.Selector('<style>', true));
+            const parsedInlineStyle = utils.parseCSS(cssText, { strict: true });
+            if (parsedInlineStyle[0]) {
+              addProps(parsedInlineStyle[0][1], new utils.Selector('<style>', true));
+            }
           }

           // store reference to an element we need to compile style="" attr for
